test/studies/diffusion/heat_2d.chpl

```diff
@@ -0,0 +1,45 @@
+// setup simulation parameters
+config const xLen : real = 2,
+             yLen : real = 2,
+             nx = 31,
+             ny = 31,
+             nt = 50,
+             sigma = 0.25,
+             nu = 0.05;
+
+const dx : real = xLen / (nx - 1),
+      dy : real = yLen / (ny - 1),
+      dt : real = sigma * dx * dy / nu;
+
+// define a 2D domain and subdomain
+const dom = {0..<nx, 0..<ny},
+      domInner : subdomain(dom) = dom.expand(-1);
+
+// define an array to hold the solution
+var u : [dom] real;
+
+// setup initial conditions
+u = 1;
+u[
+  (0.5 / dx):int..<(1.0 / dx + 1):int,
+  (0.5 / dy):int..<(1.0 / dy + 1):int
+] = 2;
+
+// run finite difference computation
+var un = u;
+for n in 0..#nt {
+  u <=> un;
```

Comment:
The swap operator is pretty unusual (and uses what other languages consider to be the spaceship operator for comparisons), so I'd put a comment here noting it (like `// swap the two arrays to set up for the next iteration`).